### ISSUE TITLE
improve travis build times

### ISSUE THREAD

**Issue by ibtaylor** [halide/Halide#2129]
Since the entire project is being built multiple times, wouldn't it be beneficial to turn on ccache in `.travis.yml`?

```
cache:
- ccache
```
I believe this should set up `$HOME/.ccache` and prepend `/usr/lib/ccache` to `PATH`.

`CXX_=g++-4.8 CC_=gcc-4.8` should still be valid since symlinks for those will exist in `/usr/lib/ccache`

**Comment by abadams** [comment]
Hrm. This is a very good point. I'll look into it.

**Comment by ibtaylor** [comment]
@abadams waiting on it to build in #2130 PR

**Comment by abadams** [comment]
Cool. I'm setting it up on our non-travis linux / arm linux buildbots at the same time.

**Comment by ibtaylor** [comment]
There is probably a way to also have it do incremental builds instead of builds completely from scratch.
Also, it seems like people configure travis to persist/restore `ccache` between builds.

A quick improvement in `build_travis.sh` could be:
```bash
DEFAULT_JOBS=2
NPROC=$(($(nproc 2> /dev/null || echo $DEFAULT_JOBS)+1))
```
which should end up being `${DEFAULT_JOBS}+1` or `number_of_cores+1` suitable for `make -j${NPROC}`
+1 is typical for including non-worker parent `make` process, I believe.


I'm not incredibly familiar with travis quiet yet, so I may be of limited help.

**Comment by kgnk** [comment]
N.B. In various projects over the years, I've seen stale build products make a project seem to build properly, but that same project fails to build in a clean workspace.

**Comment by abadams** [comment]
Good point. If we can make *just* the ccache cache persistent, that seems
safer.

But the majority of travis build latency is just waiting for the build to
get scheduled. For all in-repo PRs we prefer the buildbots at
https://buildbot.halide-lang.org/master

**Comment by abadams** [comment]
We don't use travis now
